sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -44,6 +44,9 @@ class RegexParser(pattern: String) {
   private var pos = 0
 
   def parse(): RegexAST = {
+    if (pattern == null) {
```

Comment:
Since the code seems to always pass the result of a toString on the parameter, does this happen in practice?